File: leetcode/1004.max-consecutive-ones-iii.cpp
/*
 * @lc app=leetcode id=1004 lang=cpp
 *
 * [1004] Max Consecutive Ones III
 */

// @lc code=start
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int max_count=0;
        int zero_count=0;
        int left = 0;

        for (int right = 0; right<nums.size(); right++) {
            if (nums[right]==0) {
                zero_count++;
            }
            while (zero_count>k) {
                if (nums[left]==0) {
                    zero_count-=1;
                }
                left++;
            }
            max_count=std::max(max_count, right-left+1);
        }

        return max_count;
        
    }
};
// @lc code=end

File: leetcode/1004.max-consecutive-ones-iii.py
#
# @lc app=leetcode id=1004 lang=python3
#
# [1004] Max Consecutive Ones III
#

# @lc code=start
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        maxCount = 0
        left = 0
        zero_count=0
        
        for right, num in enumerate(nums):
            # if (num==1):
            #     count+=1
            #     right+=1
            if (num==0):
                zero_count+=1
                
            while zero_count>k:
                if nums[left]==0:
                    zero_count-=1
                left+=1
                
            
            maxCount=max(maxCount, right-left+1)
        
        return maxCount
                    
                
        
# @lc code=end

